### PR TITLE
report: create naming convention for css variables

### PR DESCRIPTION
**Pull request by connorjclark**
The only visual change should be this layout fix (see that bad chevron):

Before
image
After
image

Open to discussion. Note, I will follow up with sorting. Some stuff to start us off:

1. First off, thoughts on the naming convention?
2. For variables that affect the entire report, what "component" name should be used? We kinda used `body` before, but I don't like that because places like PSI / DevTools don't provide the entire body to the report. I'm thinking either `report` or `default`.
3. Should _everything_ be prepended with `lh-`?
4. Should we be using only Material Design colors?
5. Most of these variables were simple to rename. I had trouble with some, which I marked.

### CODE REVIEW

File: lighthouse-core/report/html/report-styles.css [hunk 1]
```diff
@@ -15,41 +15,56 @@
  * limitations under the License.
  */
 
+/*
+  Naming convention:
```
Comment (patrickhulce) [inline lighthouse-core/report/html/report-styles.css]: ❤️

File: lighthouse-core/report/html/report-styles.css [hunk 2]
```diff
@@ -15,41 +15,56 @@
...
+  If a variable is used for a specific component: --{component}-{property name}-{modifier}
+
+  Both {component} and {property name} should be kebab-case. If the target is the entire page,
+  use 'report' for the component. The property name should be not be abbreviated. Use the
+  property name the variable is intended for - if it's used for multiple, a common descriptor
+  is fine (ex: 'size' for a variable applied to 'width' and 'height'). If a variable is shared
+  across multiple components, either create more variables or just drop the "{component}-"
+  part of the name. Append any modifiers at the end (ex: 'big', 'dark').
+
+  For colors: --color-{type}-{hue}
```
Comment (patrickhulce) [inline lighthouse-core/report/html/report-styles.css]: seems like you mean intensity/lightness/saturation here instead of `hue`, `type` is currently the hue :)

File: lighthouse-core/report/html/report-styles.css [hunk 3]
```diff
@@ -15,41 +15,56 @@
...
+  --category-header-font-size: 20px;
   --header-line-height: 24px;
-  --caption-line-height: 16px;
+  --audit-explanation-line-height: 16px;
   --default-padding: 12px;
-  --section-padding: 16px;
-  --section-indent: 16px;
-  --audit-item-gap: 5px;
-  --text-indent: 8px;
+  --footer-vertical-padding: 16px;
```
Comment (patrickhulce) [inline lighthouse-core/report/html/report-styles.css]: seems like we're a tad loose with `property-name` should this be `--footer-padding-vertical` according to the convention? maybe add this as a counter example to the comment if not?

File: lighthouse-core/report/html/report-styles.css [hunk 4]
```diff
@@ -213,34 +228,34 @@
...
+  --score-icon-size: 10px;
+  --score-gauge-label-font-size-big: 22px;
+  --score-gauge-label-font-size: 14px;
+  --score-gauge-label-line-height-big: 26px;
+  --score-gauge-label-line-height: 20px;
 
-  --lh-audit-vpadding: 4px;
-  --lh-audit-group-vpadding: 12px;
-  --lh-section-vpadding: 8px;
+  --lh-audit-vertical-padding: 4px;
```
Comment (patrickhulce) [inline lighthouse-core/report/html/report-styles.css]: same deal here with `padding-vertical` vs. `vertical-padding`

File: lighthouse-core/report/html/report-styles.css [hunk 5]
```diff
@@ -320,7 +335,7 @@
 }
 
 .lh-audit__description, .lh-audit__stackpack {
-  --inner-audit-right-padding: calc(var(--text-indent) + 2px);
+  --inner-audit-right-padding: var(--stackpack-horizontal-padding);
```
Comment (patrickhulce) [inline lighthouse-core/report/html/report-styles.css]: elsewhere we used `margin-right` which seemed to align with the convention, flip here too to match property name?

File: lighthouse-core/report/html/report-styles.css (hunk 2 above)
Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: Yes, thanks!

Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: I couldn't find an MD lingo for describing those tags (`700`, `A700`, ...)

File: lighthouse-core/report/html/report-styles.css (hunk 3 above)
Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: Yeah that makes sense. `padding-vertical` isn't a property, but `padding-{top,bottom}` is, and `padding-vertical` follows from using the "common descriptor" rule.

File: lighthouse-core/report/html/report-styles.css (hunk 5 above)
Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: for sure

File: lighthouse-core/report/html/report-styles.css [hunk 6]
```diff
@@ -15,41 +15,57 @@
...
+  If a variable is used for a specific component: --{component}-{property name}-{modifier}
+
+  Both {component} and {property name} should be kebab-case. If the target is the entire page,
+  use 'report' for the component. The property name should be not be abbreviated. Use the
+  property name the variable is intended for - if it's used for multiple, a common descriptor
+  is fine (ex: 'size' for a variable applied to 'width' and 'height'). If a variable is shared
+  across multiple components, either create more variables or just drop the "{component}-"
+  part of the name. Append any modifiers at the end (ex: 'big', 'dark').
+
+  For colors: --color-{type}-{intensity}
```
Comment (patrickhulce) [inline lighthouse-core/report/html/report-styles.css]: would probably go with `{hue}` over `{type}` since I'm not sure what `type` means in the context of a color, I know my last comment was kinda ambiguous on that sorry!

**Comment by patrickhulce** [comment]
Oh and this looks pre-existing but the large amount of padding on our failure audits jumped out as unusual to me when doing a quick visual check

image

**Comment by Hoten** [comment]
> Oh and this looks pre-existing but the large amount of padding on our failure audits jumped out as unusual to me when doing a quick visual check

Confirmed it exists in a different PR's now deployment, so yeah pre-existing.

image

Empty display text wants some space too.

**Comment by Hoten** [comment]
> TODOs are followups?

Would like to resolve some of them here. The ones that just say `TODO: rename` I couldn't figure out a good name for them, any ideas?

also, if we don't want to prefix all vars with `lh-` I'll remove the ones that do have it.

File: lighthouse-core/report/html/report-styles.css [hunk 7]
```diff
@@ -15,41 +15,57 @@
...
   --header-line-height: 24px;
-  --caption-line-height: 16px;
+  --audit-explanation-line-height: 16px;
   --default-padding: 12px;
-  --section-padding: 16px;
-  --section-indent: 16px;
-  --audit-item-gap: 5px;
-  --text-indent: 8px;
+  --footer-padding-vertical: 16px;
+  --audit-item-gap: 5px; /* TODO rename. */
```
Comment (patrickhulce) [inline lighthouse-core/report/html/report-styles.css]: `audit-margin-horizontal`?

File: lighthouse-core/report/html/report-styles.css [hunk 8]
```diff
@@ -15,41 +15,57 @@
...
+  --stackpack-padding-horizontal: 10px;
   --secondary-text-color: var(--color-black-800);
   --informative-color: var(--color-blue-900);
-  --medium-75-gray: #757575;
-  --medium-50-gray: hsl(210, 17%, 98%);
+  --medium-75-gray: #757575; /* TODO rename / switch to MD */
```
Comment (patrickhulce) [inline lighthouse-core/report/html/report-styles.css]: I think this is exactly `--color-gray-600`

File: lighthouse-core/report/html/report-styles.css [hunk 9]
```diff
@@ -15,41 +15,57 @@
...
+  --medium-50-gray: hsl(210, 17%, 98%); /* TODO rename / switch to MD */
```
Comment (patrickhulce) [inline lighthouse-core/report/html/report-styles.css]: `--color-gray-50: #fafafa` instead?

File: lighthouse-core/report/html/report-styles.css [hunk 10]
```diff
@@ -15,41 +15,57 @@
...
   --lh-sparkline-height: 5px;
-  --lh-audit-vpadding: 8px;
-  --lh-audit-group-vpadding: 8px;
-  --lh-section-vpadding: 12px;
+  --lh-audit-padding-vertical: 8px;
+  --lh-audit-group-padding-vertical: 8px;
+  --lh-section-padding-vertical: 12px;
   --chevron-size: 12px;
-  --inner-audit-left-padding: calc(var(--score-shape-size) + var(--score-shape-margin-left) + var(--score-shape-margin-right));
+  --inner-audit-left-padding: calc(var(--score-icon-size) + var(--score-icon-margin-left) + var(--score-icon-margin-right)); /* TODO: rename */
```
Comment (patrickhulce) [inline lighthouse-core/report/html/report-styles.css]: `--audit-description-padding-left`?

File: lighthouse-core/report/html/report-styles.css (hunk 8 above)
Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: right you are

File: lighthouse-core/report/html/report-styles.css [hunk 11]
```diff
@@ -15,41 +15,58 @@
...
-  --medium-50-gray: hsl(210, 17%, 98%);
+  --color-gray-600: #757575;
+  --color-gray-50: #FAFAFA;
+  --snippet-background-color: var(--color-gray-50);
   --report-secondary-border-color: #ebebeb;
-  --display-value-gray: hsl(216, 5%, 39%);
+  --chevron-line-stroke: hsl(216, 5%, 39%);
   --report-width: calc(60 * var(--body-font-size));
   --report-min-width: 400px;
   --report-header-height: 161px;
```
Comment (paulirish) [inline lighthouse-core/report/html/report-styles.css]: this is the only --report- prefixed item that refers to a specific component.. 


oh look at that.. it's actually outdated and old now. PR drafted: https://github.com/GoogleChrome/lighthouse/pull/9188

File: lighthouse-viewer/app/styles/viewer.css [hunk 12]
```diff
@@ -9,7 +9,7 @@
 }
 
 :root {
-  --lh-background-color: #304ffe;
```
Comment (paulirish) [inline lighthouse-viewer/app/styles/viewer.css]: seems like this isn't used anymore. i think it can die.

File: lighthouse-core/report/html/report-styles.css [hunk 13]
```diff
@@ -15,41 +15,58 @@
...
+  For colors: --color-{hue}-{intensity}
+
+  {intensity} is the Material Design tag - 700, A700, etc. Omit "-{intensity}" if there is only one
+  color of that hue.
+*/
 .lh-vars {
   --text-font-family: Roboto, Helvetica, Arial, sans-serif;
   --monospace-font-family: 'Roboto Mono', 'Menlo', 'dejavu sans mono', 'Consolas', 'Lucida Console', monospace;
   --body-background-color: #fff;
   --body-text-color: var(--color-black-900);
```
[the remark this answers is not in the document]
Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: Let's do `--report-`.

File: lighthouse-core/report/html/report-styles.css [hunk 14]
```diff
@@ -15,41 +15,58 @@
...
+  --audit-explanation-line-height: 16px;
   --default-padding: 12px;
-  --section-padding: 16px;
-  --section-indent: 16px;
-  --audit-item-gap: 5px;
-  --text-indent: 8px;
+  --footer-padding-vertical: 16px;
+  --audit-margin-horizontal: 5px;
+  --stackpack-padding-horizontal: 10px;
   --secondary-text-color: var(--color-black-800);
```
[the remark this answers is not in the document]
Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: I could go either way with this. The two ways to interpret...

1. `--secondary-text-color`, if you consider `secondary-text` to be a "component"...
2. `--color-secondary-text`, because anytime you reach out for a color it's nice to just do `--color` and look at autocomplete. The average / fail / etc. colors already are like this.

I'm leaning towards 2), plus expanding the color naming rule to cover these.

File: lighthouse-core/report/html/report-styles.css [hunk 15]
```diff
@@ -15,41 +15,58 @@
...
   --informative-color: var(--color-blue-900);
-  --medium-75-gray: #757575;
-  --medium-50-gray: hsl(210, 17%, 98%);
+  --color-gray-600: #757575;
+  --color-gray-50: #FAFAFA;
+  --snippet-background-color: var(--color-gray-50);
```
[the remark this answers is not in the document]
Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: There's actually a few of the color component variables defined in this manner already:

```
--report-text-color: var(--color-black-900);
--informative-color: var(--color-blue-900);
--snippet-background-color: var(--color-gray-50);
--tools-icon-color: var(--color-gray-600);
--env-item-background-color: var(--color-black-100);
--plugin-badge-background-color: var(--color-white);
--topbar-background-color: var(--color-black-100);
--metrics-toggle-background-color: var(--color-black-200);
```

which is kinda nice because inverting the `--color-` variables is all that needs to be done to do dark theming.

[the remark this answers is not in the document]
Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: or do you mean removing the variable `--color-gray-50` and inlining the value here

File: lighthouse-viewer/app/styles/viewer.css (hunk 12 above)
Comment (Hoten) [reply lighthouse-viewer/app/styles/viewer.css]: I'm actually gonna revert it b/c I don't want to touch the viewer at all rn.

File: lighthouse-core/report/html/report-styles.css [hunk 16]
```diff
@@ -39,7 +39,7 @@
   --report-text-color: var(--color-black-900);
   --report-font-size: 16px;
   --report-line-height: 24px;
-  --snippet-color: hsl(206, 6%, 25%);
+  --snippet-color: var(--color-black-800);
```
Comment (Hoten) [inline lighthouse-core/report/html/report-styles.css]: look at `Background and foreground colors do not have a sufficient contrast ratio.` in the now deployment for a snippet.

**Comment by Hoten** [comment]
ok, another visual change made its way in. we were not changing the cheveron color in dark theme

before

image


after

image

File: lighthouse-core/report/html/report-styles.css [hunk 17]
```diff
@@ -15,113 +15,129 @@
...
-  --color-black-100: #F5F5F5;
-  --color-black-200: #E0E0E0;
-  --color-black-400: #BDBDBD;
-  --color-black-500: #9E9E9E;
-  --color-black-600: #757575;
-  --color-black-800: #424242;
-  --color-black-900: #212121;
-  --color-black: #000000;
-  --color-blue: #2962FF;
+  --color-gray-50: #FAFAFA;
```
Comment (brendankenny) [inline lighthouse-core/report/html/report-styles.css]: I wonder how we started using `gray` instead of `grey` :)

File: lighthouse-core/report/html/report-styles.css [hunk 18]
```diff
@@ -15,113 +15,129 @@
...
+  --color-gray-400: #BDBDBD;
+  --color-gray-500: #9E9E9E;
+  --color-gray-600: #757575;
+  --color-gray-800: #424242;
+  --color-gray-900: #212121;
+  --color-gray: #000000;
+  --color-blue-A700: #2962FF;
   --color-green-700: #018642;
-  --color-green: #0CCE6B;
+  --color-green: #0CCE6B; /* TODO: this is not a MD color */
```
Comment (brendankenny) [inline lighthouse-core/report/html/report-styles.css]: > `/* TODO: this is not a MD color */`

do we care?

File: lighthouse-core/report/html/report-styles.css [hunk 19]
```diff
@@ -15,113 +15,129 @@
...
+  use 'report' for the component. The property name should be not be abbreviated. Use the
+  property name the variable is intended for - if it's used for multiple, a common descriptor
+  is fine (ex: 'size' for a variable applied to 'width' and 'height'). If a variable is shared
+  across multiple components, either create more variables or just drop the "{component}-"
+  part of the name. Append any modifiers at the end (ex: 'big', 'dark').
+
+  For colors: --color-{hue}-{intensity}
+
+  {intensity} is the Material Design tag - 700, A700, etc. Omit "-{intensity}" if there is only one
+  color of that hue.
```
Comment (brendankenny) [inline lighthouse-core/report/html/report-styles.css]: if this gets longer or really applies to multiple places it could go in CONTRIBUTING.md, but seems ok here now

File: lighthouse-core/report/html/report-styles.css [hunk 20]
```diff
@@ -15,113 +15,129 @@
...
 .lh-vars {
   --text-font-family: Roboto, Helvetica, Arial, sans-serif;
   --monospace-font-family: 'Roboto Mono', 'Menlo', 'dejavu sans mono', 'Consolas', 'Lucida Console', monospace;
-  --body-background-color: #fff;
-  --body-text-color: var(--color-black-900);
-  --body-font-size: 16px;
-  --body-line-height: 24px;
-  --subheader-color: hsl(206, 6%, 25%);
-  --header-font-size: 20px;
+  --report-background-color: #fff;
```
Comment (brendankenny) [inline lighthouse-core/report/html/report-styles.css]: since these are all changing in blame anyways, does it make sense to sort these, at least grouping by component?

May want to do that at the very end, though, cause the diff will be hard to follow

File: lighthouse-core/report/html/report-styles.css [hunk 21]
```diff
@@ -15,113 +15,129 @@
...
+  --audit-group-margin-bottom: 40px;
+  --env-item-background-color: var(--color-gray-100);
   --env-name-min-width: 220px;
-  --env-tem-padding: 10px 0px;
-  --expandable-padding: 0 0 2px calc(var(--score-shape-margin-left) + var(--score-shape-size) + var(--score-shape-margin-right));
+  --env-item-padding: 10px 0px;
+  --env-item-font-size: 28px;
+  --env-item-line-height: 36px;
+  --metric-description-padding: 0 0 2px calc(var(--score-icon-margin-left) + var(--score-icon-size) + var(--score-icon-margin-right));
   --gauge-circle-size-big: 112px;
```
Comment (brendankenny) [inline lighthouse-core/report/html/report-styles.css]: should these be `--score-gauge-*` like the other gauge vars? Or the others should be `--gauge-*`?

File: lighthouse-core/report/html/report-styles.css [hunk 22]
```diff
@@ -15,113 +15,129 @@
...
-  --score-title-font-size-big: 28px;
-  --score-title-font-size: 20px;
-  --score-title-line-height-big: 36px;
-  --score-title-line-height: 26px;
+  --score-gauge-wrapper-width: 148px;
+  --score-gauge-percentage-font-size-big: 38px;
+  --score-gauge-percentage-font-size: 28px;
+  --score-icon-margin-left: 4px;
+  --score-icon-margin-right: 12px;
+  --score-icon-margin: 0 var(--score-icon-margin-right) 0 var(--score-icon-margin-left);
```
Comment (brendankenny) [inline lighthouse-core/report/html/report-styles.css]: can we pick different prefix than score for one or both of the score things since they're quite different?

File: lighthouse-core/report/html/report-styles.css [hunk 23]
```diff
@@ -180,27 +195,27 @@
...
-    --body-font-size: 14px;
-    --body-line-height: 20px;
+    --audit-group-margin-bottom: 20px;
+    --report-font-size: 14px;
+    --report-line-height: 20px;
     --env-name-min-width: 120px;
     --gauge-circle-size-big: 96px;
     --gauge-circle-size: 72px;
     --header-padding: 16px 0 16px 0;
     --plugin-icon-size: 75%;
```
Comment (brendankenny) [inline lighthouse-core/report/html/report-styles.css]: kebab case does present kind of a problem for parsing component vs property, but that's not new here

File: lighthouse-core/report/html/report-styles.css (hunk 17 above)
Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: fwiw the MD sites all say gray. the correct way, coincidentally

File: lighthouse-core/report/html/report-styles.css (hunk 18 above)
Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: goood question. calling @yuinchien - should we only use colors from Material design?

File: lighthouse-core/report/html/report-styles.css (hunk 20 above)
Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: I want to wait until a completely separate PR, so it's in two commits and separates concerns in the PRs. Want to be able to follow through any changes in UIs like github's.

File: lighthouse-core/report/html/report-styles.css (hunk 22 above)
Comment (Hoten) [reply lighthouse-core/report/html/report-styles.css]: changed `--score-gauge` to `--gauge`

File: lighthouse-core/report/html/report-styles.css (hunk 17 above)
Comment (brendankenny) [reply lighthouse-core/report/html/report-styles.css]: > fwiw the MD sites all say gray.

ha, well the third party site in the comment above doesn't :)

> the correct way, coincidentally

freedom is the only way, yeah
